spec/lib/response-handler.spec.js

```diff
@@ -0,0 +1,134 @@
+            expect(responseHandler.isErrorCode(404)).toBe(true);
+        });
+    });
+    describe(".createErrorForResponse()", () => {
+        var errorFromResponse, errorResponse;
+
+        it("returns any unknown errors", () => {
+            errorResponse = {
+                body: {
+                    code: "anUnknownError"
+                }
+            };
+            errorFromResponse = responseHandler.createErrorForResponse(errorResponse);
+            expect(errorFromResponse.message).toBe("UNKNOWN");
+            expect(errorFromResponse.code).toBe("UNKNOWN");
+        });
+        it("returns a timeout error", () => {
+            errorResponse = {
+                code: "exampleTimeoutError"
+            };
+            errorFromResponse = responseHandler.createErrorForResponse(errorResponse);
+            expect(errorFromResponse.message).toBe("Socket Timeout");
+            expect(errorFromResponse.code).toBe("ESOCKETTIMEDOUT");
+        });
+        it("returns known errors", () => {
+            errorResponse = {
+                error: "duplicate_artifact",
+                response: {
+                    body: {
+                        code: "duplicate_artifact",
+                        message: "A duplicate artifact error message"
+                    }
+                }
+            };
+            errorFromResponse = responseHandler.createErrorForResponse(errorResponse);
+            expect(errorFromResponse.message).toBe("A duplicate artifact error message");
+            expect(errorFromResponse.code).toBe("DUPLICATE_ARTIFACT");
+        });
+    });
+    describe(".handleErrorResponse()", () => {
+        it("throws errors from responseHandler.createErrorForResponse()", () => {
+            spyOn(responseHandler, "createErrorForResponse").andReturn(new Error("SomeError"));
+            expect(() => {
+                responseHandler.handleErrorResponse();
+            }).toThrow();
+        });
+    });
+    describe(".resolveLink()", () => {
+        it("returns a new URI as a string", () => {
+            var link, resolvedLink, url;
+
+            link = "link/example";
+            url = "url/example";
+            resolvedLink = responseHandler.resolveLink(link, url);
+            expect(resolvedLink).toBe("link/url/example");
+        });
+    });
+    describe(".resolveLinkHeaders()", () => {
+        it("returns an array of urls", () => {
+            var response, urls;
+
+            response = {
+                headers: {
+                    link: "</morty/artifact/shelf-js-test/an-artifact/2016-11-23T16:03:05.239Z>; rel=\"item\"; title=\"artifact\", </morty/artifact/shelf-js-test/an-artifact/2016-11-23T16:01:29.716Z>; rel=\"item\"; title=\"artifact\", </morty/artifact/shelf-js-test/an-artifact/2016-11-30T15:00:38.132Z>; rel=\"item\"; title=\"artifact\""
+                },
+                request: {
+                    href: "https://api.shelf-qa.cwscloud.net/morty/artifact/_search"
```

Comment:
Could you not use our real host? Something like "my-shelf.example.com" would be better